README.md

```diff
@@ -221,3 +224,72 @@ Not all steps need to be followed on the instructions for the aad-pod-identity p
+
+
+## Local End-To-End Testing for the Key Vault Azure Provider
+
+This section will show you how to locally test the Azure Key Vault Provider end-to-end (e2e). The e2e tests utilize Bats for testing the scripts. Take a look inside the `test/bats/tests/local` folder to see the tests and the deployments needed for creating the e2e tests.
+
+### Prerequisites
+- [ Helm >= 3](https://helm.sh/)
+- [Kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+- [Azure Service Principal](https://docs.microsoft.com/en-us/cli/azure/create-an-azure-service-principal-azure-cli?view=azure-cli-latest)
+- [Azure Key Vault](https://docs.microsoft.com/en-us/azure/key-vault/key-vault-manage-with-cli2)
+- [Kind](https://kind.sigs.k8s.io/docs/user/quick-start/)
+- [Docker](https://docs.docker.com/get-started/)
+- [Bats](https://github.com/bats-core/bats-core)
+
+
+Here are the steps that you can follow to test the Azure Key Vault Azure Provider.
+
+1. Add your secrets to a `secrets.env` file at the application `root` directory.
+
+    💡 The third Key Vault object information is the same as the first  object. Only, the 3rd object will also include an `objectAlias`.
+    ```bash
+    # secrets.env
+    PROVIDER_TEST_IMAGE=<your_docker_username>/secrets-store-csi-driver-provider-azure
+
+    KEYVAULT_OBJECT_NAME1=<yourKeyVaultSecretName>
+    KEYVAULT_OBJECT_VALUE1=<yourKeyVaultSecretValue>
+    KEYVAULT_OBJECT_TYPE1=secret
+    KEYVAULT_OBJECT_ALIAS1=""
+    KEYVAULT_OBJECT_VERSION1=""
+
+    KEYVAULT_OBJECT_NAME2=<yourKeyVaultKeyName>
+    KEYVAULT_OBJECT_VALUE2=<yourKeyVaultKeyValue>
+    KEYVAULT_OBJECT_TYPE2=key
+    KEYVAULT_OBJECT_ALIAS2=""
+    KEYVAULT_OBJECT_VERSION2=""
+
+    # same information as 1st Key Vault Object. Just add objectAlias
+    KEYVAULT_OBJECT_NAME3=<yourKeyVaultSecretName>
+    KEYVAULT_OBJECT_VALUE3=<yourKeyVaultSecretValue>
+    KEYVAULT_OBJECT_TYPE3=secret
+    KEYVAULT_OBJECT_ALIAS3=<YOUR_KEY_VAULT_SECRET_ALIAS>
+    KEYVAULT_OBJECT_VERSION3=""
+
+    KEYVAULT_NAME=<yourAzureKeyVaultName>
+
+    AZURE_CLIENT_ID=<yourAzureServicePrincipalId>
+    AZURE_CLIENT_SECRET=<yourAzureServicePrincipalSecret>
+    TENANT_ID=<yourAzureTenantId>
+    ```
+
+2. Now, you will create a `kind cluster` and kubernetes namespace locally.
+
+    ```bash
+      # bootstrap the kind cluster
+      make local-e2e-bootstrap
+
+      # set your kubernetes context to be the kind cluster
+      export KUBECONFIG="$(kind get kubeconfig-path --name="kind")"
+
+      # creat the `dev` namespace
+      kubectl create ns dev
```

Comment:
Can the namespace creation not be part of the `local-e2e-bootstrap` target in the Makefile - will simplify the developer flow.

Reply:
This can be removed for now if we are merging the e2e targets and staying with Helm 2 for now.

When you put in another PR to perform the upgrade to Helm3, this should be moved to the `e2e-bootstrap` target in the Makefile.